src/seedu/addressbook/AddressBook.java

```diff
@@ -208,7 +208,46 @@
 
     public static void main(String[] args) {
         showWelcomeMessage();
+        
+        /* original code before SLAP reduction
         processProgramArgs(args);
+        */
+
+        /** T2A3 Exercise 1
+         * After SLAP reduction
+         */
+        if (args.length >= 2) {
+            showToUser(MESSAGE_INVALID_PROGRAM_ARGS);
+            exitProgram();
+        }
+
+        if (args.length == 1) {
+            setupGivenFileForStorage(args[0]);
+        }
+
+        if(args.length == 0) {
+            setupDefaultFileForStorage();
+        }
+        
+        /** T2A3 Exercise 2
+         * Peixuan's refactoring to make the readability even worse
+        if (args.length < 0) {
+        } else {
+            if (args.length > 2) {
+                showToUser(MESSAGE_INVALID_PROGRAM_ARGS);
+                exitProgram();
+            } else {
+                if (args.length == 1) {
+                    setupGivenFileForStorage(args[0]);
+                } else {
+                    if (args.length == 0) {
+                        setupDefaultFileForStorage();
+                    }
+                }
+            }
+        }
```

Comment:
Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.